apps/examples/testcase/ta_tc/st_things/stc/stc_st_things_main.c

```diff
@@ -0,0 +1,176 @@
+	return true;
+}
+
+static bool handle_ownership_transfer_request(void)
+{
+	printf("[%s]Received a request for Ownership-transfer. \n", TAG);
+	return true;
+}
+
+static bool handle_get_request(st_things_get_request_message_s *req_msg, st_things_representation_s *resp_rep)
+{
+	printf("Received a GET request on %s\n", req_msg->resource_uri);
+
+	if (0 == strncmp(req_msg->resource_uri, g_res_switch, strlen(g_res_switch))) {
+		return handle_get_request_on_switch(req_msg, resp_rep);
+	} else if (0 == strncmp(req_msg->resource_uri, g_res_dimming, strlen(g_res_dimming))) {
+		return handle_get_request_on_dimming(req_msg, resp_rep);
+	} else if (0 == strncmp(req_msg->resource_uri, g_res_temp, strlen(g_res_temp))) {
+		return handle_get_request_on_ct(req_msg, resp_rep);
+	} else {
+		printf("Not supported uri.\n");
+	}
+	return false;
+}
+
+static bool handle_set_request(st_things_set_request_message_s *req_msg, st_things_representation_s *resp_rep)
+{
+	printf("Received a SET request on %s\n", req_msg->resource_uri);
+
+	if (0 == strncmp(req_msg->resource_uri, g_res_switch, strlen(g_res_switch))) {
+		return handle_set_request_on_switch(req_msg, resp_rep);
+	} else if (0 == strncmp(req_msg->resource_uri, g_res_dimming, strlen(g_res_dimming))) {
+		return handle_set_request_on_dimming(req_msg, resp_rep);
+	} else if (0 == strncmp(req_msg->resource_uri, g_res_temp, strlen(g_res_temp))) {
+		return handle_set_request_on_ct(req_msg, resp_rep);
+	}
+	return false;
+}
+
+/**
+* @testcase         stc_st_things_initialize_p
+* @brief            Initialize st_things stack 
+* @scenario         Initialize st_things stack 
+* @apicovered       st_things_initialize 
+* @precondition     none
+* @postcondition    none
+*/
+static void stc_st_things_initialize_p(void)
+{
+	int res;
+	bool easysetup_complete = false;	
+	res = st_things_initialize("test_device_def.json", &easysetup_complete);
+	TC_ASSERT_EQ("st_things_initialize", res, ST_THINGS_ERROR_NONE);
+	TC_SUCCESS_RESULT();
+}
+
+
+/**
+* @testcase         stc_st_things_initialize_n
+* @brief            Initialize st_things stack 
+* @scenario         Initialize st_things stack 
+* @apicovered       st_things_initialize 
+* @precondition     none
+* @postcondition    none
+*/
+static void stc_st_things_initialize_n(void)
+{
+	int res;
+	bool easysetup_complete = false;
+	res = st_things_initialize("test_device_def.json", &easysetup_complete);
+	res = st_things_initialize("test_device_def.json", &easysetup_complete);
+	TC_ASSERT_EQ("st_things_initialize", res, ST_THINGS_ERROR_STACK_ALREADY_INITIALIZED);
+	TC_SUCCESS_RESULT();
+}
+
+
+#ifdef CONFIG_BUILD_KERNEL
+int main(int argc, FAR char *argv[])
+#else
+int stc_st_things_main(int argc, char *argv[])
+#endif
+{
+	if (tc_handler(TC_START, "ST_THINGS STC") == ERROR) {
+		return ERROR;
+	}
+
+	/* Positive TCs */
+	stc_st_things_initialize_p();
+	stc_st_things_initialize_n();
+	return 0;
+}
+
+int st_things_stc(int argc, FAR char *argv[])
```

Comment:
Why do you need this function?
Where do you call this function?

Reply:
Removed.  Will be added in future if required.